python/hsfs/engine/python.py

```diff
@@ -479,7 +488,15 @@ def _split_df(self, df, splits):
         random.shuffle(groups)
         df[split_column] = groups
         for i, item in enumerate(items):
-            result_dfs[item[0]] = df[df[split_column] == i].drop(split_column, axis=1)
+            split_df = df[df[split_column] == i].drop(split_column, axis=1)
+            if item == training_dataset_obj.train_split:
+                transformation_function_engine.TransformationFunctionEngine.populate_builtin_transformation_functions(
+                    training_dataset_obj, feature_view_obj, split_df
+                )
+            result_dfs[item[0]] = self._apply_transformation_function(
```

Comment:
If first `item[0]` is not `training_dataset_obj.train_split`, this does not work.

Reply:
I mean first `item` may not guarantee to be `train_split`?

Reply:
`item[0]` is just a tuple with split name and ratio. here we are just constructing dict  `result_dfs` which is split name and dataframe. Not sure why it has to be  `train_split`?